cmd/seccomp.go

```diff
@@ -0,0 +1,137 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	apiv1 "k8s.io/api/core/v1"
+	k8sRuntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+func checkSeccomp(resource k8sRuntime.Object, result *Result) {
+	annotations := getPodAnnotations(resource)
+	podAnnotation := apiv1.SeccompPodAnnotationKey
+	podVal, podOk := annotations[podAnnotation]
+
+	if podOk {
+		if badSeccompProfileName(podVal) {
+			occ := Occurrence{
+				container: "",
+				id:        ErrorSeccompDisabledPod,
+				kind:      Error,
+				message:   fmt.Sprintf("Seccomp disabled for pod."),
+				metadata: Metadata{
+					"Reason": prettifyReason(podVal),
+				},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+			return
+		}
+
+		if podVal == apiv1.DeprecatedSeccompProfileDockerDefault {
+			occ := Occurrence{
+				container: "",
+				id:        ErrorSeccompDeprecatedPod,
+				kind:      Warn,
+				message:   fmt.Sprintf("Seccomp annotation for pod set to deprecated value."),
+				metadata: Metadata{
+					"Reason": prettifyReason(podVal),
+				},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+		}
+	}
+
+	for _, container := range getContainers(resource) {
+		containerAnnotation := apiv1.SeccompContainerAnnotationKeyPrefix + container.Name
+		val, ok := annotations[containerAnnotation]
+
+		if !ok && !podOk {
+			occ := Occurrence{
+				container: container.Name,
+				id:        ErrorSeccompAnnotationMissing,
+				kind:      Error,
+				message:   fmt.Sprintf("Seccomp annotation missing."),
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+			continue
+		}
+
+		if ok && badSeccompProfileName(val) {
+			occ := Occurrence{
+				container: container.Name,
+				id:        ErrorSeccompDisabled,
+				kind:      Error,
+				message:   fmt.Sprintf("Seccomp disabled for container."),
+				metadata: Metadata{
+					"Reason": prettifyReason(val),
+				},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+			continue
+		}
+
+		if ok && val == apiv1.DeprecatedSeccompProfileDockerDefault {
+			occ := Occurrence{
+				container: container.Name,
+				id:        ErrorSeccompDeprecated,
+				kind:      Warn,
+				message:   fmt.Sprintf("Seccomp annotation for container set to deprecated value."),
+				metadata: Metadata{
+					"Reason": prettifyReason(val),
+				},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+		}
+	}
+}
+
+func badSeccompProfileName(val string) bool {
+	switch {
+	case val == apiv1.SeccompProfileRuntimeDefault:
+		return false
+	case val == apiv1.DeprecatedSeccompProfileDockerDefault:
+		return false
+	case strings.HasPrefix(val, ProfileNamePrefix):
```

Comment:
Yes I did forget! Will fix

Reply:
Hm actually I think this is ok... It will return false if the profile starts with `localhost/` and returning false means it is an ok profile name. There is a test for all three of these cases. Is there a way I can make the code less confusing / easier to read?

Reply:
Nope, I might just not have had enough ☕️ this morning.